lib/_stream_readable.js

```diff
@@ -300,9 +300,7 @@ function chunkInvalid(state, chunk) {
 // 'readable' event will be triggered.
 function needMoreData(state) {
   return !state.ended &&
-         (state.needReadable ||
-          state.length < state.highWaterMark ||
-          state.length === 0);
+         (state.needReadable || state.length < state.highWaterMark);
```

Comment:
Can `state.highWaterMark` be `0`? If so, `<` should probably be `<=`.

Reply:
this should be `<=`.

Reply:
We should probably add a test for this case.